cloud-agent/service/server/src/main/scala/org/hyperledger/identus/iam/authorization/keycloak/admin/KeycloakPermissionManagementService.scala

```diff
@@ -28,60 +27,58 @@ case class KeycloakPermissionManagementService(
   override def grantWalletToUser(
       walletId: WalletId,
       entity: KeycloakEntity
-  ): ZIO[WalletAdministrationContext, PermissionManagement.Error, Unit] = {
+  ): ZIO[WalletAdministrationContext, PermissionManagementServiceError, Unit] = {
     for {
       _ <- walletManagementService
         .findWallet(walletId)
         .someOrFail(WalletNotFoundById(walletId))
 
       walletResourceOpt <- findWalletResource(walletId)
-        .logError("Error while finding wallet resource")
-        .mapError(UnexpectedError.apply)
 
       walletResource <- ZIO
         .fromOption(walletResourceOpt)
         .orElse(createWalletResource(walletId))
-        .logError("Error while creating wallet resource")
-        .mapError(UnexpectedError.apply)
       _ <- ZIO.log(s"Wallet resource created ${walletResource.toString}")
 
       permission <- createResourcePermission(walletResource.getId, entity.id.toString)
-        .mapError(UnexpectedError.apply)
 
       _ <- ZIO.log(s"Permission created with id ${permission.getId} and name ${permission.getName}")
     } yield ()
   }
 
-  private def createResourcePermission(resourceId: String, userId: String): Task[UmaPermissionRepresentation] = {
+  private def createResourcePermission(resourceId: String, userId: String): UIO[UmaPermissionRepresentation] = {
     val policy = UmaPermissionRepresentation()
     policy.setName(policyName(userId, resourceId))
     policy.setUsers(Set(userId).asJava)
 
     for {
-      umaPermissionRepresentation <- ZIO.attemptBlocking(
-        authzClient
-          .protection()
-          .policy(resourceId)
-          .create(policy)
-      )
+      umaPermissionRepresentation <- ZIO
+        .attemptBlocking(
+          authzClient
+            .protection()
+            .policy(resourceId)
+            .create(policy)
+        )
+        .orDie
     } yield umaPermissionRepresentation
   }
 
-  private def findWalletResource(walletId: WalletId): Task[Option[ResourceRepresentation]] = {
+  private def findWalletResource(walletId: WalletId): UIO[Option[ResourceRepresentation]] = {
     for {
       walletResource <- ZIO
         .attemptBlocking(
-          authzClient.protection().resource().findById(walletId.toUUID.toString())
+          authzClient.protection().resource().findById(walletId.toUUID.toString)
         )
         .asSome
         .catchSome { case e: RuntimeException =>
-          if (e.getMessage().contains("Could not find resource")) ZIO.none
+          if (e.getMessage.contains("Could not find resource")) ZIO.none
```

Comment:
@patlo-iog Just a side comment, but this looks rather fragile... 🧐

Reply:
Indeed. It's a shame library we use don't expose information about resource not found :smiling_face_with_tear:, but I can recheck.